src/form/FileUpload/FileList/FileList.tsx

```diff
@@ -74,49 +114,71 @@ const FileListItem: React.FC<FileListItemProps> = ({
 	onCancel,
 	fileUploadErrorLabel,
 	fileUploadInProgressLabel,
-}: FileListItemProps) => {
+}) => {
+	const [preview, setPreview] = useState<string>(file.preview || '');
+
 	const isUploading = isFileUploading(file);
-	const classes = classNames({
-		'file-list-item--is-uploading': isUploading,
-	});
+	const isIndeterminate = isFileUploadingIndeterminate(file);
+	const { read, readerCleanup } = useFileReader(setPreview);
+
+	useEffect(() => {
+		return () => {
+			readerCleanup();
+		};
+	}, []);
+
+	useEffect(() => {
```

Comment:
May I suggest this useEffect is moved into the `useFileReader` hook? I think that would make the implementation a lot cleaner and more reusable.

Reply:
good tip thx